src/main/java/org/broadinstitute/hellbender/engine/TwoPassVariantWalker.java

```diff
@@ -0,0 +1,90 @@
+package org.broadinstitute.hellbender.engine;
+
+import htsjdk.variant.variantcontext.VariantContext;
+import org.broadinstitute.hellbender.engine.filters.CountingReadFilter;
+import org.broadinstitute.hellbender.engine.filters.VariantFilter;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+
+import java.util.stream.StreamSupport;
+
+/**
+ * A VariantWalker that makes two passes through the variants.
+ * This allows the user to store internal states during the first pass, which the user can then
+ * process and access during the second pass
+ **/
+public abstract class TwoPassVariantWalker extends VariantWalker {
+    /**
+     * Overrides the default, single-pass traversal framework of {@link VariantWalkerBase}
+     */
+    @Override
+    public void traverse(){
+        final VariantFilter variantContextFilter = makeVariantFilter();
+        final CountingReadFilter readFilter = makeReadFilter();
+
+        // First pass through the variant
+        traverseVariants(variantContextFilter, readFilter, this::firstPassApply);
+        logger.info("Finished first pass through the variants");
+
+        // Process the data accumulated during the first pass
+        afterFirstPass();
+
+        // Second pass
+        logger.info("Starting second pass through the variants");
+        traverseVariants(variantContextFilter, readFilter, this::secondPassApply);
+    }
```

Comment:
Add `logger.info(readFilter.getSummaryLine())` after the second traversal pass.

Reply:
done